routing/blinding.go

```diff
@@ -124,15 +133,38 @@ func NewBlindedPaymentPathSet(paths []*BlindedPayment) (*BlindedPaymentPathSet,
 	// replacement, so our target pub key in this case just remains the
 	// real introduction node ID.
 	for _, path := range paths {
-		if len(path.BlindedPath.BlindedHops) != 1 {
-			continue
-		}
+		pathLength := len(path.BlindedPath.BlindedHops)
 
-		pathSet = []*BlindedPayment{path}
-		finalCLTVDelta = path.CltvExpiryDelta
-		targetPub = path.BlindedPath.IntroductionPoint
+		if pathLength == 1 {
+			pathSet = []*BlindedPayment{path}
+			finalCLTVDelta = path.CltvExpiryDelta
+			targetPub = path.BlindedPath.IntroductionPoint
+
+			break
+		}
 
-		break
+		// We add a dummy hop to the end of the path so that
+		// we can do MPP path finding taking all the blinded
+		// routes into account. Moreover we need to use a
+		// dummy hop here because we still want to be able to
+		// penalize the last hop in case the payment fails.
+		// We need to set the cipher text because it is used for the
+		// payload size estimation.
+		lastHopCipherText := path.BlindedPath.BlindedHops[pathLength-1].
```

Comment:
can the cipher text be smaller, just to allow longer routes (very hypothetical)?

Reply:
added a carve-out in the pathfinder:

```
		// If the edge points to the NUMS key we don't need to consider
		// the payload size again because we already added the size of
		// the dummy hop which will be removed after the route is found.
		if fromVertex != source &&
			!isBlindedRouteNUMSTargetKey(ToPubKey[:]) {
```

wdyt ?

Reply:
or do you think it is too expensive to do this check for every pathfinding edge and we should just accept this additional hop for blinded paths, I kinda lean towards the latter and accept this additional hop in size.

Reply:
right, I wanted to check how expensive this is, I'd also tend to not add this check here

Reply:
I also thought about a different approach, setting the ciphertext to 0 for the dummy hop, but the payload size would be not complete accurate as well without intervention into the code, because only the final hop accounts for amount and cltv delta for blinded paths, so I tend to remove it and go with this additional hop.

Reply:
let's see what @ellemouton thinks about it.

Reply:
why cant the payload be zero length?

Reply:
removed the carve-out and added the logic to the size-function and the lasthop payload function.